ykllvmwrap/src/jitmodbuilder.cc

```diff
@@ -520,49 +548,83 @@ class JITModBuilder {
       // `deoptimize` call so the stopgap interpreter can access it.
       Value *YKCPArg = JITFunc->getArg(JITFUNC_ARG_INPUTS_STRUCT_IDX);
       LiveVals.push_back(YKCPArg);
-      std::tuple<size_t, size_t, Instruction *> YkCPAlloca = getYkCPAlloca();
+      std::tuple<size_t, size_t, Instruction *, size_t> YkCPAlloca =
+          getYkCPAlloca();
       AOTMap[YKCPArg] = YkCPAlloca;
 
       IntegerType *Int32Ty = Type::getInt32Ty(Context);
       PointerType *Int8PtrTy = Type::getInt8PtrTy(Context);
 
-      // Create struct storing current basic block index and instruction index.
-      // This will be needed later to point the stopgap interpeter at the
-      // correct location from where to start interpretation.
+      // Create struct type for storing a vector and its length.
+      PointerType *VecPtrTy = PointerType::get(Context, 0);
+      StructType *VecLenStructTy =
+          StructType::get(Context, {VecPtrTy, PointerSizedIntTy});
+
+      // Create a vector of active stackframes (i.e. basic block index,
+      // instruction index, function name). This will be needed later to point
+      // the stopgap interpeter at the correct location from where to start
+      // interpretation and to setup its stackframes.
       // FIXME: Use function index instead of string name.
-      StructType *CurPosSTy = StructType::get(
+      ActiveFrames.push_back(
+          {CurBBIdx, CurInstrIdx, FuncName}); // Add current frame.
+      StructType *ActiveFrameSTy = StructType::get(
           Context, {PointerSizedIntTy, PointerSizedIntTy, Int8PtrTy});
-      AllocaInst *CurPos = FailBuilder.CreateAlloca(
-          CurPosSTy, ConstantInt::get(PointerSizedIntTy, 1));
-      auto GEP = FailBuilder.CreateGEP(CurPosSTy, CurPos,
+      AllocaInst *ActiveFrameVec = FailBuilder.CreateAlloca(
+          ActiveFrameSTy,
+          ConstantInt::get(PointerSizedIntTy, ActiveFrames.size()));
+      for (size_t I = 0; I < ActiveFrames.size(); I++) {
+        std::tuple<size_t, size_t, StringRef> FrameInfo = ActiveFrames[I];
+        size_t BBIdx = std::get<0>(FrameInfo);
+        size_t InstrIdx = std::get<1>(FrameInfo);
+        StringRef Fn = std::get<2>(FrameInfo);
+
+        auto GEP =
+            FailBuilder.CreateGEP(ActiveFrameSTy, ActiveFrameVec,
+                                  {ConstantInt::get(PointerSizedIntTy, I),
+                                   ConstantInt::get(Int32Ty, 0)});
```

Comment:
Why `Int32Ty`?

Reply:
Good question. This can also be `PointerSizedIntTy` I guess. In this case I know that my struct doesn't have more than i32 fields so it could even be something much smaller (e.g. i8).

Reply:
I'm not sure what "my struct ... fields" means here. What is this integer actually indexing?

Reply:
This GEP instruction makes an index into a vector of `ActiveFrameVec` structs, where the struct looks like this:
```
struct ActiveFrameVec {
    bbidx: usize,
    instridx: usize,
    fname: *const i8,
}
```
So the first `ConstantInt(PointerSizedIntTy, I)` indexes into the vector, while the second `ConstantInt::get(Int32Ty, 0)` indexes into the first field of that struct. Since this struct only has 3 fields I could change the `Int32Ty` to an `Int8Ty`.

Reply:
Ah! Maybe this bit of code just needs a short comment to make this clear (it never hurts to comment why code is correct in this sort of case!).

Reply:
Let's try this: 8d593502